File: Parents/parents.cs
using System;

namespace ConsoleApplication
{
    public class Program
    {
        public struct parentChild
        {
            public string parent;
            public string child;

        }
        public static int findString(string[] list, int x, string target)
        {
            for (int i = 0; i < x; i++)
            {
                if (list[i] == target)
                {
                    return i;
                }
                return -1;
            }
        }
        public static void sort(string[] list, string[] list2, int x)
        {
            for (int i = 0; i < x-1; i++)
            {
                int y = x;
                for (int z = y + 1; z < x; z++)
                {
                    if (list[z] < list[y])
                    {
                        y=z;
                    }
                }
                string t = list[y];
                list[y] = list[x];
                list[x] = t;
                t = list2[y];
                list2[y] = list2[i];
                list2[x] = t;
            }
        }
        public static void Main(string[] args)
        {
            
            Console.WriteLine("This program prints parent-child pairs.\nEnter parents and children below, use 'quit' to stop.");
            string[] child = new string[100];
            string[] parent = new string[100];
            int count  = 0;

            while (true)
            {
                Console.Write("Parent: ");
                string new_parent,child,name;
                {
                    if (new_parent == "quit")
                    {
                        break;
                    }
                    int i = findString(child,count,new_parent);
                    if (i < 0)
                    {
                        child[count] = new_parent;
                        parent[count] = "";
                        count++;
                    }
                    Console.Write("Children of " + new_parent + ": ");
                    child = Console.ReadLine();
                    while(true)
                    {
                        i = findString(child, count, name);
                        if (i < 0)
                        {
                            child[count] = name;
                            i = count++;
                        }
                        parent[i] = new_parent;
                    }

                }
            }
            sort(child,parent,count);
            Console.WriteLine();
            Console.Write("Child    Parent\n");
            Console.Write("-----    ------\n");
            for (int i = 0; i < length; i++)
            {
                
            }
        }
    }
}

File: Parents/parents.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int FindString(string list[],int n,string target)
{
    for (int i=0; i < n; i++)
       if (list[i]==target)
           return i;
    return -1;
}

void Sort(string list[],string list2[],int n)
{
    for (int i=0; i < n-1; i++) {
       int j = i;
       for (int k=j+1; k < n; k++)
           if (list[k] < list[j])
              j = k;
       string t = list[j]; 
       list[j] = list[i]; 
       list[i] = t;
       t = list2[j];
       list2[j] = list2[i];
       list2[i] = t;
    }
}

int main()
{
    cout << "This program prints parent-child pairs." << endl;
    cout << "Enter parents and children below, use 'quit' to stop." << endl;
    
    string child[100];
    string parent[100];
    int count = 0;

    while (true) {
       cout << "Parent: ";
       string new_parent,children,name;
       getline(cin,new_parent);    
       if (new_parent == "quit") break;
       
        int i = FindString(child,count,new_parent);
        if (i < 0) {
            child[count] = new_parent;
            parent[count] = "";
            count++;
        }
       
       cout << "Children of " << new_parent << ": ";
       getline(cin,children);
       stringstream ss(children);
       while (ss >> name) {
           i = FindString(child,count,name);
           if (i < 0) {
               child[count] = name;
               i = count++;
           }
           parent[i] = new_parent;
       }
    }
    
    Sort(child,parent,count);
    
    cout << endl;
    cout << "Child    Parent" << endl;
    cout << "-----    ------" << endl;
    for (int i=0; i < count; i++)
       cout << setw(9) << left << child[i] << parent[i] << endl;
}
